kotlinx-coroutines-core/common/src/CoroutineScope.kt

```diff
@@ -117,13 +123,30 @@ public fun MainScope(): CoroutineScope = ContextScope(SupervisorJob() + Dispatch
 /**
  * Returns `true` when the current [Job] is still active (has not completed and was not cancelled yet).
  *
- * Check this property in long-running computation loops to support cancellation:
+ * A coroutine cancallation [is cooperative](https://kotlinlang.org/docs/cancellation-and-timeouts.html#cancellation-is-cooperative)
+ * and normally, the cancellation machinery happens when the coroutine *suspends*, for example,
+ * when trying to read from a [channel][kotlinx.coroutines.channels.Channel] that is empty.
+ *
+ * Sometimes, a coroutine does not need to do any of such operations, but still wants to be a cooperative
```

Comment:
```suggestion
 * Sometimes, a coroutine does not need to do any of such operations, but still wants to be cooperative
```

This is a funny typo!

> cooperative (noun)
> an enterprise or organization owned by and operated for the benefit of those using its services

Who doesn't want to be one?